src/components/sectionList/download/DownloadDialog.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button, ButtonStrip, Modal, ModalActions } from '@dhis2/ui'
+import React from 'react'
+import { LinkButton } from '../../LinkButton'
+import { DownloadDialogContent } from './DownloadDialogContent'
+import { DownloadFormWrapper } from './DownloadForm'
+import { useDownloadUrl } from './useDownloadUrl'
+
+type DownloadDialogProps = {
+    onClose: () => void
+    selectedModels: Set<string>
+}
+
+export const DownloadDialog = ({
+    onClose,
+    selectedModels,
+}: DownloadDialogProps) => {
+    return (
+        <Modal onClose={onClose}>
+            <DownloadFormWrapper>
+                <DownloadDialogContent selectedModels={selectedModels} />
+                <Actions onClose={onClose} selectedModels={selectedModels} />
+            </DownloadFormWrapper>
+        </Modal>
+    )
+}
+
+const Actions = ({ onClose, selectedModels }: DownloadDialogProps) => {
+    const downloadUrl = useDownloadUrl({ selectedModels })
+    return (
+        <ModalActions>
+            <ButtonStrip>
+                <Button onClick={onClose} secondary>
+                    Cancel
+                </Button>
+                <LinkButton primary onClick={onClose} href={downloadUrl}>
```

Comment:
This is done using `download=true` to the server. We could do both I guess? But it's a bit safer to let the server handle it, because then it's easier to catch in eg. a test if the server doesn't behave as it should.

Reply:
Hmm not sure what's correct tbh.. I guess it can't harm us doing both? Not sure if that's correct though